File: VB/WindowsApplication1/RepositoryItemMyMemoEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.Accessibility
Imports System.Drawing

Namespace WindowsApplication1
    <UserRepositoryItem("RegisterMyMemoEdit")> _
    Public Class RepositoryItemMyMemoEdit
        Inherits RepositoryItemMemoEdit
        Public Event Assigned As EventHandler

        Shared Sub New()
            RegisterMyMemoEdit()
        End Sub

        Public Sub New()
            fScrollWidth = 17
            fScrollButtonWidth = 17
        End Sub

        Public Const MyMemoEditName As String = "MyMemoEdit"
        Public Overrides ReadOnly Property EditorTypeName() As String
            Get
                Return MyMemoEditName
            End Get
        End Property

        Public Shared Sub RegisterMyMemoEdit()
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(MyMemoEditName, GetType(MyMemoEdit), GetType(RepositoryItemMyMemoEdit), GetType(MyMemoEditViewInfo), New MemoEditPainter(), True, New Bitmap(16, 16), GetType(TextEditAccessible)))
        End Sub

        Public Overrides Sub Assign(ByVal item As RepositoryItem)
            BeginUpdate()
            Try
                MyBase.Assign(item)
                Dim source As RepositoryItemMyMemoEdit = TryCast(item, RepositoryItemMyMemoEdit)
                If source Is Nothing Then
                    Return
                End If
                fScrollWidth = source.fScrollWidth
                fScrollButtonWidth = source.fScrollButtonWidth
                RaiseEvent Assigned(Me, New EventArgs())
            Finally
                EndUpdate()
            End Try
        End Sub

        Private fScrollWidth As Integer
        Private fScrollButtonWidth As Integer

        <DefaultValue(17)> _
        Public Property ScrollWidth() As Integer
            Get
                Return fScrollWidth
            End Get
            Set(ByVal value As Integer)
                If fScrollWidth <> value Then
                    fScrollWidth = value
                    OnPropertiesChanged()
                    If OwnerEdit IsNot Nothing Then
                        TryCast(OwnerEdit, MyMemoEdit).UpdateScrollBars()
                    End If
                End If
            End Set
        End Property

        <DefaultValue(17)> _
        Public Property ScrollButtonWidth() As Integer
            Get
                Return fScrollButtonWidth
            End Get
            Set(ByVal value As Integer)
                If fScrollButtonWidth <> value Then
                    fScrollButtonWidth = value
                    OnPropertiesChanged()
                    If OwnerEdit IsNot Nothing Then
                        TryCast(OwnerEdit, MyMemoEdit).UpdateScrollBars()
                    End If
                End If
            End Set
        End Property
    End Class

End Namespace

File: CS/WindowsApplication1/RepositoryItemMyMemoEdit.cs
using System;
using System.ComponentModel;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Drawing;
using DevExpress.Accessibility;

namespace WindowsApplication1
{
    [UserRepositoryItem("RegisterMyMemoEdit")]
    public class RepositoryItemMyMemoEdit : RepositoryItemMemoEdit
    {
        public virtual event EventHandler Assigned;
        
        static RepositoryItemMyMemoEdit() { RegisterMyMemoEdit(); }

        public RepositoryItemMyMemoEdit()
        {
            fScrollWidth = 17;
            fScrollButtonWidth = 17;
        }

        public const string MyMemoEditName = "MyMemoEdit";
        public override string EditorTypeName { get { return MyMemoEditName; } }

        public static void RegisterMyMemoEdit()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(
                MyMemoEditName,                     //Specifies the editor's name.
                typeof(MyMemoEdit),                 //Specifies the type of the editor class.
                typeof(RepositoryItemMyMemoEdit),   //Specifies the type of the custom repository item class.
                typeof(MyMemoEditViewInfo),         //Specifies the type of the editor's ViewInfo class.
                new MemoEditPainter(),              //An instance of the editor's Painter class.
                true,                               //Specifies whether the custom editor will be visible at design time within container controls
                null,                               //Specifies the image that will be displayed along
                typeof(TextEditAccessible)
                ));
        }
        
        public override void Assign(RepositoryItem item)
        {
            BeginUpdate();
            try
            {
                base.Assign(item);
                RepositoryItemMyMemoEdit source = item as RepositoryItemMyMemoEdit;
                if (source == null) return;
                fScrollWidth = source.fScrollWidth;
                fScrollButtonWidth = source.fScrollButtonWidth;
                Assigned(this, new EventArgs());
            }
            finally
            {
                EndUpdate();
            }
        }

        private int fScrollWidth;
        private int fScrollButtonWidth;

       [DefaultValue(17)]
        public int ScrollWidth
        {
            get { return fScrollWidth; }
            set
            {
                if (fScrollWidth != value)
                {
                    fScrollWidth = value;
                    OnPropertiesChanged();
                    if (OwnerEdit != null)
                        (OwnerEdit as MyMemoEdit).UpdateScrollBars();
                }
            }
        }

        [DefaultValue(17)]
        public int ScrollButtonWidth
        {
            get { return fScrollButtonWidth; }
            set
            {
                if (fScrollButtonWidth != value)
                {
                    fScrollButtonWidth = value;
                    OnPropertiesChanged();
                    if (OwnerEdit != null)
                        (OwnerEdit as MyMemoEdit).UpdateScrollBars();
                }
            }
        }
    }

}
